tests/unit_tests/audit_log/test_with_python_runner.py

```diff
@@ -77,33 +83,160 @@ def factory(
+    while True:
+        i += 1
+        yield f"UUID-{id}-{i}"
+
+
+def expect_count_rows(
+    table_name: TableName,
+    query: str,
+) -> list[list[str, MockResultSet]]:
+    def count_rows(event_name: str) -> list[str, MockResultSet]:
+        return expect_query(
+            f"""
+            INSERT INTO {table_name.fully_qualified} (
+              "LOG_TIMESTAMP",
+              "ROW_COUNT",
+              "SESSION_ID",
+              "DB_OBJECT_NAME",
+              "DB_OBJECT_SCHEMA",
+              "DB_OBJECT_TYPE",
+              "EVENT_ATTRIBUTES",
+              "EVENT_NAME",
+              "LOG_SPAN_ID",
+              "LOG_SPAN_NAME",
+              "RUN_ID"
+            ) SELECT
+              SYSTIMESTAMP(),
+              (SELECT count(1) FROM "S2"."T"),
+              CURRENT_SESSION,
+              'T',
+              'S2',
+              'TABLE',
+              '{{{{"a": 123, "b": "value"}}}}',
+              '{event_name}',
+              'UUID-2-2',
+              'INSERT',
+              'UUID-2-1'\u0020
+            """
+        )
+
+    return [
+        count_rows("Begin"),
+        expect_query(query),
+        count_rows("End"),
+    ]
 
 
-def test_audit(aaf_pytest_db_schema, context_mock):
-    sql_executor = MockSQLExecutor()
-    sample = SampleInput()
-    child_qh_factory=StartFinishQh
-    with audit_query_handler_factory(
+def expect_query_with_temp_view(
+    db_schema: str,
+    view_name: str,
+    query: str,
+    column: Column,
+) -> list[list[str, MockResultSet]]:
+    view_fq = ViewNameImpl(view_name, SchemaName(db_schema)).fully_qualified
+    return [
+        expect_query(
+            f"""
+            CREATE OR REPLACE VIEW {view_fq} AS
+            {query} as {column.name.name};
+            """
+        ),
+        expect_query(
+            f"""
+            SELECT
+                "{column.name.name}"
+            FROM {view_fq};
+            """,
+            MockResultSet(rows=[(1,)], columns=[column]),
+        ),
+        expect_query(f"DROP VIEW IF EXISTS {view_fq};"),
+    ]
+
+
+@patch("exasol.analytics.audit.audit.uuid.uuid4")
+@patch("exasol.analytics.query_handler.query.select.uuid.uuid4")
+def test_audit(uuid_mock_1, uuid_mock_2, aaf_pytest_db_schema, prefix, context_mock):
+    uuid_mock_1.side_effect = uuid_generator("1")
+    uuid_mock_2.side_effect = uuid_generator("2")
+    sql_executor = create_sql_executor(
         aaf_pytest_db_schema,
-        child_qh_factory=child_qh_factory,
-    ) as factory:
-        runner = PythonQueryHandlerRunner[SampleInput, SampleOutput](
-            sql_executor=sql_executor,
-            top_level_query_handler_context=context_mock,
-            parameter=sample,
-            query_handler_factory=factory,
-        )
-        output = runner.run()
-    assert output.result == sample
+        prefix,
+        expect_query(""),
+        expect_query(""),
+    )
+    audit_table = AuditTable(aaf_pytest_db_schema, prefix, run_id=Mock())
```

Comment:
you are mixing the prefix for tmp db objects with the table_prefix for the audit log, that is usually not a good idea. From the usage, you can probably use an arbitrary string for the table_prefix,